pkg/storage/mysql/mysql_test.go

```diff
@@ -1,21 +1,113 @@
-package mysql_test
+package mysql
 
 import (
+	"context"
 	"testing"
+	"time"
 
-	"github.com/openfga/openfga/pkg/storage/common"
-	"github.com/openfga/openfga/pkg/storage/mysql"
+	sq "github.com/Masterminds/squirrel"
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/openfga/openfga/pkg/storage/sqlcommon"
 	"github.com/openfga/openfga/pkg/storage/test"
 	storagefixtures "github.com/openfga/openfga/pkg/testfixtures/storage"
+	"github.com/openfga/openfga/pkg/tuple"
 	"github.com/stretchr/testify/require"
+	openfgav1 "go.buf.build/openfga/go/openfga/api/openfga/v1"
 )
 
 func TestMySQLDatastore(t *testing.T) {
 	testDatastore := storagefixtures.RunDatastoreTestContainer(t, "mysql")
 
 	uri := testDatastore.GetConnectionURI()
-	ds, err := mysql.New(uri, common.NewConfig())
+	ds, err := New(uri, sqlcommon.NewConfig())
 	require.NoError(t, err)
 	defer ds.Close()
 	test.RunAllTests(t, ds)
 }
+
+// TestReadEnsureNoOrder asserts that the read response is not ordered by ulid
+func TestReadEnsureNoOrder(t *testing.T) {
+	testDatastore := storagefixtures.RunDatastoreTestContainer(t, "mysql")
+
+	uri := testDatastore.GetConnectionURI()
+	ds, err := New(uri, sqlcommon.NewConfig())
+	require.NoError(t, err)
+
+	ctx := context.Background()
+	defer ds.Close()
+
+	store := "store"
+	objectType := "doc"
+	objectID1 := "object_id_1"
+	relation := "relation"
+	user1 := "user:user_1"
+	firstTuple := tuple.NewTupleKey(objectType+":"+objectID1, relation, user1)
```

Comment:
Here, and elsewhere, this code can be simplified quite a bit to simply

```go
store := "store"
relation := "relation"
firstTuple := tuple.NewTupleKey("doc:1", relation, "user:1")
secondTuple := tuple.NewTupleKey("doc:2", relation, "user:2")

... // inner code

// we expect that objectID1 will return first because it is inserted first
curTuple, err := iter.Next()
require.NoError(t, err)
require.Equal(t, firstTuple, curTuple.Key)

curTuple, err = iter.Next()
require.NoError(t, err)
require.Equal(t, secondTuple, curTuple.Key)
```

Reconstructing the TupleKey objects and string concatenating the objects is not necessary here.

Reply:
done

Reply:
You changed the last bits, but you didn't simplify at the top.

```go
store := "store"
firstTuple := tuple.NewTupleKey("doc:1", "relation", "user:1")
secondTuple := tuple.NewTupleKey("doc:2", "relation", "user:2")
```